daemon/images/image_builder.go

```diff
@@ -161,7 +167,27 @@ func (i *ImageService) pullForBuilder(ctx context.Context, name string, authConf
 	if err := i.pullImageWithReference(ctx, ref, platform, nil, pullRegistryAuth, output); err != nil {
 		return nil, err
 	}
-	return i.GetImage(name, platform)
+
+	img, err := i.GetImage(name, platform)
+	if errdefs.IsNotFound(err) && img != nil && platform != nil {
+		imgPlat, parseErr := platforms.Parse(path.Join(img.OS, img.BaseImgArch(), img.BaseImgVariant()))
+		if parseErr != nil {
+			return nil, err
+		}
+
+		p := *platform
+		if !platforms.Only(p).Match(imgPlat) {
+			po := streamformatter.NewJSONProgressOutput(output, false)
+			progress.Messagef(po, "", `
+WARNING: Pulled image with specified platform (%s), but the resulting image's configured platform (%s) does not match.
+This is most likely caused by a bug in the build system that created the fetched image (%s).
+Please notify the image author to correct the configuration.
+				`, platforms.Format(p), platforms.Format(imgPlat), name)
+			logrus.WithError(err).WithField("image", name).Warn("Ignoring error about platform mismatch where the manifest list points to an image whose configuration does not match the platform in the manifest.")
```

Comment:
Think it's worth logging image platform as well?

Reply:
We can, but it is in the error message itself which we already have.

Reply:
ah, yes, makes sense